### ISSUE TITLE
No Docs

### ISSUE THREAD

**Issue by ZariMatan** [magento/devdocs#1548]
Feedback on page: /guides/v2.0/rest/list.html

Hey,

I see a lot of differences between what I see here and what is seen [here](http://devdocs.magento.com/swagger/#/).

It's almost impossible to understand the what to send in every request or if the request is available.

**Comment by keharper** [comment]
The Swagger docs you linked to are for version 2.2, while the page you referenced is version 2.0, so there will be some significant differences. 

We acknowledge that Swagger provides too much and too little information. The code isn't sufficiently annotated. By default, the Model Schema section of each POST and PUT operation displays what can be sent in. You can click on Model  to get an idea of what's required and optional, but so much will depend on what you're trying to do.

I suggest taking a look at http://devdocs.magento.com/guides/v2.2/get-started/order-tutorial/order-intro.html to get an idea of how you can use some of the most important REST calls.

**Comment by keharper** [comment]
@ZariMatan Do you have any specific questions?

**Comment by ZariMatan** [comment]
Yes. I looked at the tutorial you sent, and there is no example for setting payment details on a cart.
Do you have any other resources I can use?
Thanks.

**Comment by keharper** [comment]
@ZariMatan 

The `POST /V1/carts/mine/payment-information` call sets payment details. Step 7 of the tutorial provides a simple example, but since it's showing an offline payment , the `additional_data` array was skipped. 

Each payment processor will have its own requirements for what the `additional_data` array must contain. I don't have that information. However, a credit card payment will typically include fields like 

```
cc_cid
cc_type
cc_exp_year
cc_exp_month
cc_number
```

I hope this sets you on the right path.

**Comment by ZariMatan** [comment]
I think this solves it for me.
Do you have more information on what those fields should be?
Most of them I understand, but what's cc_type, cc_number and cc_cid?
Thanks.

**Comment by keharper** [comment]
`cc_type` - The band of credit card. Typical values include `VI` (Visa), `MC` (MasterCard), `AE` (American Express), `DI` (Discover)
`cc_number` - The number on the credit card. 16 digits on a Visa card, for example
`cc_cid` - Here's an explanation: https://www.cvvnumber.com/

**Comment by ZariMatan** [comment]
Thanks Kevin, all this information is very helpful.
It there a detailed documentation where I can see everything?

**Comment by keharper** [comment]
@ZariMatan The detailed information you're asking for isn't currently available. We're looking for ways to provide this information, because this is a major gap in our documentation. Thanks for raising these questions.

**Comment by ZariMatan** [comment]
Hey Kevin,

I'm trying to implement what you suggested, and I get a general error: "An error occurred on the server. Please try to place the order again."

This is after I received a few 400's when the data I sent wasn't correct.
I tried to dig a bit deeper into the logs but I can't find anything, not in var/report and not in /var/log/exception.log.
I saw that a few other people faced this issue, but I couldn't find a solution.
This is what I send to POST -> rest/V1/.../payment-information:
{
            email: details.billing.email,
            "paymentMethod": {
                "method": "payeezy",
                additional_data: {
                    // cc_cid: details.card.cvv,
                    cc_type: type,
                    cc_exp_year: details.card.year,
                    cc_exp_month: details.card.month,
                    cc_number: details.card.number,
                }
            },
            "billing_address": {
                country_id: details.billing.countryId,
                street: [
                    details.billing.address
                ],
                company: details.billing.company,
                telephone: details.billing.phone,
                city: details.billing.city,
                firstname: details.billing.firstName,
                lastname: details.billing.lastName,
                postcode: details.billing.zipCode,
                email: details.billing.email,
                same_as_billing: 0,
                save_in_address_book: null
            },
        }

Any idea why I receive the error?
Thanks.

**Comment by ZariMatan** [comment]
Hey,

This is what I send:
{
            email: details.billing.email,
            paymentMethod: {
                method: "payeezy",
                additional_data: {
                    cc_cid: details.card.cvv,
                    cc_type: type,
                    cc_exp_year: details.card.year,
                    cc_exp_month: details.card.month,
                    cc_number: details.card.number,
                },
            },
            billing_address: {
                country_id: details.billing.countryId,
                street: [
                    details.billing.address
                ],
                company: details.billing.company,
                telephone: details.billing.phone,
                city: details.billing.city,
                firstname: details.billing.firstName,
                lastname: details.billing.lastName,
                postcode: details.billing.zipCode,
                email: details.billing.email,
                same_as_billing: 0,
                save_in_address_book: null
            },
        }

If you see anything wrong with the data please let me know, I stuck on this for some time now.
Thanks!

**Comment by jcalcaben** [comment]
@keharper what's the status on this issue?

**Comment by ZariMatan** [comment]
Hey,

It's been some time since I worked on this, and from what I remember this
was an issue with the payment gateway plugin and not Magento.
Anyway, thanks for the help!
